drivers/ethernet/eth.h

```diff
@@ -42,4 +43,39 @@ static inline void gen_random_mac(uint8_t *mac_addr, uint8_t b0, uint8_t b1, uin
 	sys_rand_get(&mac_addr[3], 3U);
 }
 
+static inline int net_eth_mac_load(const struct net_eth_mac_config *cfg, uint8_t *mac_addr)
+{
+	if (cfg == NULL || cfg->type == NET_ETH_MAC_DEFAULT) {
+		return -ENODATA;
+	}
+
+	/* Copy the static part */
+	memcpy(mac_addr, cfg->addr, cfg->addr_len);
+
+	if (cfg->type == NET_ETH_MAC_STATIC && cfg->addr_len == NET_ETH_ADDR_LEN) {
```

Comment:
if type is `NET_ETH_MAC_STATIC` and ``cfg->addr_len != NET_ETH_ADDR_LEN`` init shouldn't fail, the user could still use the api to set another mac address.